marine/marine.py

```diff
@@ -233,6 +331,51 @@ def _parse_output(output: POINTER(c_char_p), length: int) -> List[Optional[str]]
             for i in range(length)
         )
 
+    @classmethod
+    def _load_marine_packet(cls, field):
+        children = field.children
+        if children:
+            return cls._load_child_fields(field.children.contents)
+        else:
+            return cls._load_field_value(field.value)
+
+    @classmethod
+    def _load_child_fields(cls, children):
+        child_fields = {}
+        for child in children.data[: children.len]:
+            name = child.name.decode("ascii")
```

Comment:
Is it really always ascii? Probably yes, but let's make sure :)

Reply:
Changed to `utf-8`.

Reply:
If this is slow in benchmarks, consider adding a cache :)